src/sentry/models/groupopenperiod.py

```diff
@@ -98,8 +98,28 @@ def close_open_period(
             logger.warning("Open period is already closed", extra={"group_id": self.group.id})
             return
 
+        # We can't resolve issues before they started; if we've been asked to do so, the
+        # timestamp is likely from a different source than the date_started timestamp, and
+        # we should aim to fix that, so we warn.
+        # We ensure there's a valid close time so we can close the period as requested.
+        close_time = resolution_time
+        if resolution_time < self.date_started:
+            # There may be more externally accurate close times, but since we're closing it
+            # now, timezone.now() is arguably always a reasonable choice.
+            close_time = timezone.now()
```

Comment:
🤔 not 100% sure we should close in these scenarios. curious why we are getting into these states in general? a little nervous that we could be getting something out of order, and we're closing a period that wasn't meant to be closed yet.